extensions-core/druid-pac4j/src/main/java/org/apache/druid/security/pac4j/Pac4jFilter.java

```diff
@@ -96,24 +97,24 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
           "/",
           true, false, false, null);
     } else {
-      Object uid = securityLogic.perform(
+      OidcProfile profile = (OidcProfile) securityLogic.perform(
```

Comment:
can you use `UserProfile` interface that the iterator is returning?

Reply:
I'm not certain, I don't know java, but I think it's useful in here - since we use OIDC config here, the `UserProfiles` should always be complient with `OidcProfile` anyway. And later `OidcProfile` is more convenient to pass the context - `UserProfile` doesn't have explicit function to return all attributes at once, which is the goal here.

I'm eager to learn if what I've wrote is not correct, let me know.

Reply:
`profiles.iterator().next()` is returning `UserProfile` which is an interface. I was thinking that you use that but it doesn't have `getAttributes` method. But now I am wondering why not pass the whole profile itself. 

```
AuthenticationResult authenticationResult = new AuthenticationResult(profile.getId(), authorizerName, name, ImmutableMap.of("profile", profile))
```

Reply:
let's do the above and cast the result of `.perform` to UserProfile.

Reply:
Hm, okay. That sounds reasonable.